src/Version/Service/VersionBinaryService.php

```diff
@@ -65,21 +67,29 @@ public function downloadAsset(
     /**
      * @throws AccessDeniedException|NotFoundException|InvalidElementTypeException
      */
-    public function streamImage(
+    public function streamThumbnailImage(
         int $id,
         UserInterface $user
     ): StreamedResponse {
         $version = $this->repository->getVersionById($id);
-        $element = $this->repository->getElementFromVersion($version, $user);
-        if (!$element instanceof Asset\Image) {
-            throw new InvalidElementTypeException($element->getType());
+        $image = $this->repository->getElementFromVersion($version, $user);
+        if (!$image instanceof Asset\Image) {
+            throw new InvalidElementTypeException($image->getType());
+        }
+
+        $config = Config::getPreviewConfig();
```

Comment:
Should we introduce a new resolver here? Or you think its not necessary?

Reply:
Ah yes, totally forgot about that for the review!